ext/dom/parentnode.c

```diff
@@ -181,6 +181,10 @@ xmlNode* dom_zvals_to_fragment(php_libxml_ref_obj *document, xmlNode *contextNod
 					return NULL;
 				}
 
+				if (nodesc > 1) {
```

Comment:
We might need a comment here to explain why the copy is necessary, with a reference to the libxml docs talking about the merging of TEXT nodes (in which case our node is freed). I also wonder if we can't check the type of the node to avoid some unnecessary copies.

Reply:
If the number of parameters is greater than or equal to three, xmlNewDocText function always returns an address that point to $doc->documentElement->firstChild.

Reply:
I have added comment in the code.

Reply:
Thanks for the changes, I was just busy reviewing and extensively testing this right now. :)